mlx/transforms.cpp

```diff
@@ -262,6 +262,7 @@ void eval(const std::vector<array>& outputs) {
     auto stream = arr.primitive().stream();
     std::vector<std::shared_future<void>> arr_deps;
     for (auto& in : arr.inputs()) {
+      // TODO that's a bug
```

Comment:
This is unrelated to this PR, but I believe the below is a race condition (input's primitive can be wiped before the output task is scheduled). It's only relevant for multi-stream stuff. Will have to test and fix but makes sense for a follow up.